### PR TITLE
docs: auto update docs

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
* Created a script that automatically reads botonic packages readmes and bump them into the documentation.
* Run the created script to update the documentation automatically. (Last commit)

**e.g:** Watson, dashbot, ... were not documented
Screenshot 2020-11-05 at 13 01 44

## Context
Right now, we were manually copy-pasting the READMEs for every package into docs, this has lots of inconvenients:
- Hard to maintain (time, effort)
- Possibility to not have the READMEs up to date (it can be discouraging for incoming developers)
- Reduce human errors

## Approach taken / Explain the design
Created a script that consists of three parts:
1. Bumping READMEs from botonic packages to plugins documentation (generating a pre-formatted header)
2. Updating `sidebars.json` object to show these options available in docs menus.
3. Updating `i18en.json` to maintain the routing for the previous options.

Once the script has been executed, prettier will automatically format these files.

## To document / Usage example
From within website project, run:
`$ npm run update_plugins_docs`
also, it will be called automatically everytime we build the project to distribute (`$ npm run build`)

### CODE REVIEW

**Review by ericmarcos — CHANGES_REQUESTED** [review]
That's great!! Nice work.

Just one caveat: the tendency to put things under "utils". Utils are very generic subroutines that can clearly be used in many different parts of a codebase. However, in this PR, everything in `util/doc-structure.ts` has a very clear and defined scope, can't see it being used for other purposes other than updating plugin docs. Same for `util/formatting.ts` > `generateHeader` function. Only the filesystem functions and the `capitalize` one are generic enough to fit in a "utils" category. I'd move the others to the same `scripts/update-plugin-docs.ts` where they're used. If the file is too long, then split the functions in their own file under a folder `scripts/update-plugins-docs` with `index.ts`, `updateSidebars.ts`, etc

File: docs/website/scripts/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const packagesPath = resolve(join(cwd, '..', '..', 'packages'))
+const pluginsDocsPath = resolve(join(cwd, '..', 'docs', 'plugins'))
+const pluginsInfo = bumpPackageReadmesToPluginsDocs(
+  packagesPath,
+  pluginsDocsPath
+)
+const sidebarsPath = join(cwd, 'sidebars.json')
+updateSidebars(sidebarsPath, pluginsInfo)
+const i18ENPath = join(cwd, 'i18n', 'en.json')
+updatei18EN(i18ENPath, pluginsInfo)
```
Comment (ericmarcos) [inline docs/website/scripts/index.ts:20]: As I mentioned before, this should be inside a folder `scripts/update_plugins_docs/index.ts` (or similar).

**Review by ericmarcos — CHANGES_REQUESTED** [review]

File: docs/website/scripts/index.ts (hunk 1 above)
Comment (vanbasten17) [reply docs/website/scripts/index.ts]: for sure i didn't understand it right, everything simplified and moved into the corresponding folder 👍

File: docs/website/scripts/update-plugins-docs/bump-readmes-to-docs.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import { join } from 'path'
+import { PluginInfo } from '.'
+import { readDir, readFile, writeFile } from './file-system'
+
+const capitalize = (string: string) =>
```
Comment (dpinol) [inline docs/website/scripts/update-plugins-docs/bump-readmes-to-docs.ts:5]: name it toCamelCase?

File: docs/website/scripts/update-plugins-docs/bump-readmes-to-docs.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  PLUGIN_NLU_LOWERCASE: 'nlu',
+  PLUGIN_NLU_UPPERCASE: 'NLU',
+}
+
+const getPluginName = (packageName: string): string =>
+  packageName.split(PACKAGES.BOTONIC_PLUGIN_REF)[1]
+
+const extractPluginInfo = (packageName: string) => {
+  const id = getPluginName(packageName)
+  const title = getPluginName(packageName)
```
Comment (dpinol) [inline docs/website/scripts/update-plugins-docs/bump-readmes-to-docs.ts:22]: you can use id here, right?

File: docs/website/scripts/update-plugins-docs/file-system.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { readFileSync, readdirSync, writeFileSync } from 'fs'
+
+const UTF8: BufferEncoding = 'utf-8'
+
+export const readJSON = (filePath: string): void =>
```
Comment (dpinol) [inline docs/website/scripts/update-plugins-docs/file-system.ts:5]: ":void" is wrong
Can you configure .pre-commit & github CI to validate this scripts?

File: docs/website/scripts/update-plugins-docs/update-docs-references.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  pluginsInfo: PluginInfo[]
+) => {
+  const sidebarsJSON = readJSON(sidebarsPath)
+  sidebarsJSON[SIDEBARS.PLUGINS][SIDEBARS.PLUGIN_LIBRARY] = pluginsInfo.map(
+    p => `${PLUGINS_SEPARATOR}${p.id}`
+  )
+  writeJSON(sidebarsPath, sidebarsJSON)
+}
+
+export const updatei18EN = (i18ENPath: string, pluginsInfo: PluginInfo[]) => {
```
Comment (dpinol) [inline docs/website/scripts/update-plugins-docs/update-docs-references.ts:26]: updateI18en

**Review by dpinol — APPROVED** [review]
:clap:

**Review by ericmarcos — APPROVED** [review]
GAS

File: docs/website/scripts/update-plugins-docs/file-system.ts (hunk 4 above)
Comment (vanbasten17) [reply docs/website/scripts/update-plugins-docs/file-system.ts]: I have set it up! It's working correctly 👍

**Pull request merged**